spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/signalfx/SignalFxProperties.java

```diff
@@ -39,6 +39,11 @@
 	 */
 	private String uri;
 
+	/**
+	 * Uniquely identifies the app instance that is publishing metrics to SignalFx. Defaults to the local host name.
```

Comment:
I don't like those unpredictable defaults (we have a `default` concept in the metadata that is meant for that and I'd rather use it than putting that information in the description).

Is there anyway we can relax this?

Reply:
There are other pieces of the portfolio that have a similar requirement. It'd be good to consider what they currently do and align if possible. Spring Cloud Bus is the project that comes to mind as it uses a unique ID for message routing. Can you give us some pointers please, @dsyer?

Reply:
I asked @spencergibb about this as well, and I believe we use host name in discovery.

Reply:
Spring Cloud Bus uses the following:

```
"${vcap.application.name:${spring.application.name:${spring.application.name:application}}}:${vcap.application.instance_index:${spring.application.index:${local.server.port:${server.port:0}}}}:${vcap.application.instance_id:${random.value}}"
```

Note the use of `random.value` at the end. I think that makes it unsuitable for this purpose as the id will potentially change across restarts.

Reply:
The Bus id is structured as `{app/host}:{port/index}:{guid}`. If Spring Boot was supplying the first two we could add the “random” bit. Or maybe the whole thing would be useful for micrometer (don’t know).

Reply:
@wilkinsona I wouldn't worry about change across restart. Every scaling event/deployment in AWS results in a new instance id, and we tagged metrics with those instance ids.

Reply:
It's too late in the day to be doing anything sophisticated here. We'll stick with Micrometer's default for 2.0 rather than trying to be clever and override it with something else and probably getting it wrong.

Reply:
Agreed. It also doesn't really matter that much. The "source" bit that some of these monitoring systems require almost seem to be a cherry pick from the days of hierarchical metrics. Like the Bus id reveals: identification of a unique app instance is a multi-variate problem best solved with tags.